core/src/main/java/io/grpc/internal/ProxyDetectorImpl.java

```diff
@@ -188,6 +198,22 @@ private static InetSocketAddress overrideProxy(String proxyHostPort) {
     return InetSocketAddress.createUnresolved(parts[0], port);
   }
 
+  private static Supplier<ProxySelector> proxySelectorSupplier(final Proxy proxy) {
+    return new Supplier<ProxySelector>() {
+      @Override public ProxySelector get() {
+        return new ProxySelector() {
+          @Override public List<Proxy> select(URI uri) {
+            return ImmutableList.of(proxy);
```

Comment:
We avoid immutablelist (and similar collections) to reduce our method count on Android. That may not be necessary any more, but we've not been hurting that badly without them.